### PR TITLE
Namespace package

### PR DESCRIPTION
**Pull request by maxfischer2781**
This change turns the top-level ``cobald`` package into a namespace package (see issue #23). It also  changes unittests and documentation to reflect this.

### CODE REVIEW

File: docs/source/custom/package.rst [hunk 1]
```diff
@@ -1,17 +1,82 @@
-==========================
-Packaging and Distribution
-==========================
+=================================
+Using and Distributing Extensions
+=================================
 
-Ideally, any custom code is made available as a regular Python package.
+Extensions for :py:mod:`cobald` are regular Python code accessible to the interpreter.
```
Comment (eileen-kuehn) [inline docs/source/custom/package.rst]:
```suggestion
Extensions for :py:mod:`cobald` is regular Python code accessible to the interpreter.
```

File: docs/source/custom/package.rst [hunk 2]
```diff
@@ -1,17 +1,82 @@
...
+For specific problems, extensions can be defined directly in a Python configuration file.
+General purpose and reusable code should be made available as a regular Python package.
```
Comment (eileen-kuehn) [inline docs/source/custom/package.rst]: Why is it important that it is a *regular* Python package. Are there any other options?

```suggestion
General purpose and reusable code should be made available as a Python package.
```

File: docs/source/custom/package.rst [hunk 3]
```diff
@@ -1,17 +1,82 @@
...
+
+    # custom Controller implementation
+    class StaticController(Controller):
+        """Controller that sets demand to a fixed value"""
+        def __init__(self, target, demand):
+            super().__init__(target)
+            self.target.demand = demand
+
+    # use custom Controller
+    pipeline = StaticController.s(demand=50) >> DrawLineHook.s() >> CpuPool(interval=1)
```
Comment (eileen-kuehn) [inline docs/source/custom/package.rst]: Please consider, if Pools should be initialised similar to the Controllers, e.g. `CpuPool.s(interval=1)`.

File: docs/source/custom/package.rst [hunk 4]
```diff
@@ -1,17 +1,82 @@
...
+
+    # my_controller.py
+    from cobald.interfaces import Controller
+
+    class StaticController(Controller):
+        def __init__(self, target, demand):
+            super().__init__(target)
+            self.target.demand = demand
+
+Packages can be temporarily mounted via ``PYTHONPATH`` or permanently installed.
```
Comment (eileen-kuehn) [inline docs/source/custom/package.rst]: Is it correct here to talk about *mounting*? I feel like *loading* might be better?

```suggestion
Packages can be temporarily loaded via ``PYTHONPATH`` or permanently installed.
```

File: docs/source/custom/package.rst [hunk 5]
```diff
@@ -1,17 +1,82 @@
...
+Once available, packages can be imported and used in any configurations.
```
Comment (eileen-kuehn) [inline docs/source/custom/package.rst]:
```suggestion
Once available, packages can be imported and used in any configuration.
```

File: docs/source/custom/package.rst [hunk 6]
```diff
@@ -1,17 +1,82 @@
...
+.. code:: python3
+
+    #/etc/cobald/my_demo.py
+    from my_controller import StaticController
+
+    from cobald_demo.cpu_pool import CpuPool
+    from cobald_demo.draw_line import DrawLineHook
+
+    # use custom Controller from package
+    pipeline = StaticController.s(demand=50) >> DrawLineHook.s() >> CpuPool(interval=1)
```
Comment (eileen-kuehn) [inline docs/source/custom/package.rst]: Same as above.

File: docs/source/custom/package.rst [hunk 7]
```diff
@@ -1,17 +1,82 @@
...
+
+Configuration files are easy to use and modify, but impractical for reusable extensions.
+
+Python Packages
+===============
+
+For generic extensions, Python packages simplify distribution and reuse.
+Packages are individual `.py` files or folders containing several `.py` files;
+in addition, packages contain meta-data for dependency management and installation.
```
Comment (eileen-kuehn) [inline docs/source/custom/package.rst]:
```suggestion
in addition, packages contain metadata for dependency management and installation.
```

File: docs/source/custom/package.rst (hunk 4 above)
Comment (maxfischer2781) [reply docs/source/custom/package.rst]:
```suggestion
Packages can be temporarily accessed via ``PYTHONPATH`` or permanently installed.
```

File: docs/source/custom/package.rst (hunk 3 above)
Comment (maxfischer2781) [reply docs/source/custom/package.rst]: I've opened a new ticket #32 to address this. IMO this is useful to have, but not as part of this pull request.

File: docs/source/custom/overview.rst [hunk 8]
```diff
@@ -1,14 +1,15 @@
...
+========================================
+Custom Controllers, Pools and Extensions
+========================================
 
 The :py:mod:`cobald.daemon` is capable of loading any modules and code importable by its Python interpreter.
-This gives significant freedom in how to organize and implement custom functionality.
-However, the outward appearance of any plugins should reflect the :py:mod:`cobald` design.
+Any :py:class:`~.Controller`, :py:class:`~.Pool` or :py:class:`~.Decorator`
+is a regular class implementing the respective interface.
+This gives significant freedom in how to organize and implement custom extensions.
```
[the remark this answers is not in the document]
Comment (maxfischer2781) [reply docs/source/custom/overview.rst]: I've changed the wording of both sentences to be less vague.

**Pull request merged**